src/main/java/teammates/ui/automated/InstructorCourseJoinEmailWorkerAction.java

```diff
@@ -44,7 +47,21 @@ public void execute() {
         // saving the new instructor takes more time in GAE.
         // The instructor `id` can be constructed back using (instructorEmail%courseId)
         // because instructors' email cannot be changed before joining the course.
-        InstructorAttributes instructor = logic.getInstructorById(courseId, instructorEmail);
+        RetryManager rm = new RetryManager(5);
```

Comment:
@LiHaoTan I would like your input on what a good `maxDelayinSec` would be for this case

Reply:
First fail, wait for 1s, second fail, wait for 2s, third fail, wait for 4s, forth fail (no retry as 8s > 5s).

I think should be enough and maybe you could set `maxDelayInS`to 4s (multiple of 2).